velox/exec/Task.cpp

```diff
@@ -688,6 +706,16 @@ void Task::start(uint32_t maxDrivers, uint32_t concurrentSplitGroups) {
   }
 }
 
+void Task::checkExecutionMode(ExecutionMode mode) {
+  VELOX_CHECK(
+      mode == mode_,
+      fmt::format(
+          "Inconsistent task execution mode. Cannot execute in '{}' mode "
+          "for a '{}' mode task",
+          executionModeToString(mode),
+          executionModeToString(mode_)));
+}
+
```

Comment:
Can you also add the check on execution mode in createAndStartDrivers in addition to the null check on the inline executor dynamic cast? Thanks!

Reply:
createAndStartDrivers() is a private method and only used in Task::start(). Do we still want it there? What is the purpose?

Reply:
I think the current code assume createAndStartDrivers is called under parallelized execution mode. There is a check if executor is inline executor or not in that function. It is better to also explicitly check execution mode.